src/main/java/io/benlewis/tagminigame/configuration/ConfigWrapper.java

```diff
@@ -0,0 +1,74 @@
+package io.benlewis.tagminigame.configuration;
+
+import io.benlewis.tagminigame.TagPlugin;
+import org.bukkit.configuration.InvalidConfigurationException;
+import org.bukkit.configuration.file.FileConfiguration;
+import org.bukkit.configuration.file.YamlConfiguration;
+
+import java.io.File;
+import java.io.IOException;
+
+public class ConfigWrapper implements IConfigWrapper {
+
+    private final TagPlugin plugin;
+    private final String filePath;
+    private FileConfiguration config;
+
+    public ConfigWrapper(TagPlugin plugin, String filePath) {
+        this.plugin = plugin;
+        this.filePath = filePath;
+        config = reload();
+    }
+
+    public String getFilePath() {
+        return filePath;
+    }
+
+    @Override
+    public FileConfiguration getFileConfiguration() {
+        return config;
+    }
+
+    @Override
+    public FileConfiguration reload() {
+        File file = new File(plugin.getDataFolder(), getFilePath());
+        if (!file.exists()) {
+            // Attempt to copy default template of config, if one is present in jar
```

Comment:
Could extract into own method, makes reload more readable and removes need for explanatory comment.